src/olympia/lib/akismet/tasks.py

```diff
@@ -14,3 +17,10 @@ def submit_to_akismet(report_ids, submit_spam):
 def comment_check(report_ids):
     reports = AkismetReport.objects.filter(id__in=report_ids)
     return [report.comment_check() for report in reports]
+
+
+@task
+def save_akismet_report(xml):
+    reports = object_deserialize("xml", smart_str(xml))
```

Comment:
Pretty surprised this actually works, tbh.

Reply:
As discussed on IRC: I think it's actually safer to use `json` here. XML serialization is nice because it deals with custom types out of the box, but on the other hands fails badly with unicode control chars.